lambda/rest-endpoints/src/helpers.ts

```diff
@@ -10,3 +10,22 @@ export function getBodyContentAsJson(body:unknown): string {
     throw new Error("Did not recognise the body content as json or json-like");
   }
 }
+
+export function validateDateParam(dateParam:string):Date|null {
```

Comment:
Could we not use e.g. `new Date("YYYY-MM-DD")` and then assert it's not `typeof InvalidDate`? The `Date` constructor uses [`Date.parse`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse), which [uses a date time string format that's compatible IIUC](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date#date_time_string_format)

Reply:
Interesting question.... we discussed ease of use vs strict validation vs library/core and decided that there wasn't much in it. Interestingly the Date.parse logic seems to validate in the same way as this code.... i.e. 31st Feb _is_ treated as a valid date... the 2nd (or 3rd) March, depending on leap-year